.github/workflows/pip-build-osx.yml

```diff
@@ -41,11 +41,16 @@ jobs:
           cd build
           cmake -DCMAKE_BUILD_TYPE=Release -DPython_ADDITIONAL_VERSIONS=3 ..
           cd src/python
-          python setup.py bdist_wheel
+          python -m build
```

Comment:
You are right. Strange how `python -m pip install build` fixes it...
It does seem so as on my manylinux docker image:
```bash
$PYTHON311/bin/python -m pip install --user virtualenv
virtualenv -p $PYTHON311/bin/python mytest
virtualenv  -p $PYTHON311/bin/python mytest
source mytest/bin/activate
python -m pip list
# Package    Version
# ---------- -------
# pip        22.3.1
# setuptools 65.6.3
# wheel      0.38.4
```
Maybe because on windows and osx, the python used is the one from the system...

By the way, do you advise me to add `build` in pyproject.toml ?

Reply:
Note that `python -m build` is building inside a virtual env only on Linux. I hope there will be no inconsistency between what CMake finds (numpy includes for instance) and the one in the virtual env.

Reply:
> Note that `python -m build` is building inside a virtual env only on Linux.

Uh, weird, is that stated somewhere in the doc? Is virtualenv installed on the mac/win machines we are using?

> I hope there will be no inconsistency between what CMake finds (numpy includes for instance) and the one in the virtual env.

(or we can use `-n` to disable the virtual env)
An inconsistency would mean that `gudhi.__debug_info__` (or a similar string) would advertise a version of numpy which is not the one actually used during compilation? Or CMake may decide to enable a feature, which will then fail to build with a different version of the dependency?

Reply:
Done on [2ee4284](https://github.com/GUDHI/gudhi-devel/pull/769/commits/2ee42848aba062b891c133eb358d6b105fa5bc67)
I tried delvewheel, but it did not work. I will try later on a windows VM.